mkdocs/gh_deploy.py

```diff
@@ -5,19 +5,25 @@
 log = logging.getLogger(__name__)
 
 
-def gh_deploy(config):
+def gh_deploy(config, commit_message=None):
 
     if not os.path.exists('.git'):
         log.info('Cannot deploy - this directory does not appear to be a git '
                  'repository')
         return
 
-    log.info("Copying '%s' to `gh-pages` branch and pushing to GitHub.",
-             config['site_dir'])
+    command = ['ghp-import', '-p', config['site_dir']]
+
+    remote_branch = config.get('remote_branch', 'gh-pages')
+    command.extend(['-b', config['remote_branch']])
```

Comment:
Oh! The remote branch functionality will be currently broken as a side effect of #450. The old CLI parser had a one to one relationship between config settings and CLI arguments. This highlights the lack of test coverage in a few places like this.

So - we should add a remote branch argument like this PR adds message. @mbacho would you be happy to do that? Otherwise I can add it afterwards - I did break it after all :smile: